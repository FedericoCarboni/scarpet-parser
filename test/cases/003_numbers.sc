// Numbers test
12345;
123.45;
123e12;
123e+12;
123e-12