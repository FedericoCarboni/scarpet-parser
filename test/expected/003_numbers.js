/** @type {import('../../lib/Parser.js').Node} */
export const root = {
    kind: 'BinaryExpression',
    operator: ';',
    lvalue: {
        kind: 'BinaryExpression',
        operator: ';',
        lvalue: {
            kind: 'BinaryExpression',
            operator: ';',
            lvalue: {
                kind: 'BinaryExpression',
                operator: ';',
                lvalue: {
                    kind: 'NumberLiteral',
                    literal: '12345',
                    value: 12345n,
                    start: {
                        offset: 16,
                        line: 1,
                        character: 0,
                    },
                    end: {
                        offset: 21,
                        line: 1,
                        character: 5,
                    },
                },
                rvalue: {
                    kind: 'NumberLiteral',
                    literal: '123.45',
                    value: 123.45,
                    start: {
                        offset: 23,
                        line: 2,
                        character: 0,
                    },
                    end: {
                        offset: 29,
                        line: 2,
                        character: 6,
                    },
                },
                comment: '// Numbers test\n',
                start: {
                    offset: 16,
                    line: 1,
                    character: 0,
                },
                end: {
                    offset: 29,
                    line: 2,
                    character: 6,
                },
            },
            rvalue: {
                kind: 'NumberLiteral',
                literal: '123e12',
                value: 123000000000000n,
                start: {
                    offset: 31,
                    line: 3,
                    character: 0,
                },
                end: {
                    offset: 37,
                    line: 3,
                    character: 7,
                },
            },
            comment: null,
            start: {
                offset: 16,
                line: 1,
                character: 0,
            },
            end: {
                offset: 37,
                line: 3,
                character: 7,
            },
        },
        rvalue: {
            kind: 'NumberLiteral',
            literal: '123e+12',
            value: 123000000000000n,
            start: {
                offset: 39,
                line: 4,
                character: 0,
            },
            end: {
                offset: 46,
                line: 4,
                character: 7,
            },
        },
        comment: null,
        start: {
            offset: 16,
            line: 1,
            character: 0,
        },
        end: {
            offset: 46,
            line: 4,
            character: 7,
        },
    },
    rvalue: {
        kind: 'NumberLiteral',
        literal: '123e-12',
        value: 1.23e-10,
        start: {
            offset: 48,
            line: 5,
            character: 0,
        },
        end: {
            offset: 55,
            line: 5,
            character: 7,
        },
    },
    comment: null,
    start: {
        offset: 16,
        line: 1,
        character: 0,
    },
    end: {
        offset: 55,
        line: 5,
        character: 7,
    },
};
