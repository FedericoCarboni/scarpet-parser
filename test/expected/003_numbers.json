{
  "errors": [],
  "warnings": [],
  "root": {
    "kind": "BinaryExpression",
    "operator": ";",
    "lvalue": {
      "kind": "BinaryExpression",
      "operator": ";",
      "lvalue": {
        "kind": "BinaryExpression",
        "operator": ";",
        "lvalue": {
          "kind": "BinaryExpression",
          "operator": ";",
          "lvalue": {
            "kind": "NumberLiteral",
            "literal": "12345",
            "value": "12345",
            "start": {
              "pos": 16,
              "row": 1,
              "col": 0
            },
            "end": {
              "pos": 21,
              "row": 1,
              "col": 5
            }
          },
          "rvalue": {
            "kind": "NumberLiteral",
            "literal": "123.45",
            "value": 123.45,
            "start": {
              "pos": 23,
              "row": 2,
              "col": 0
            },
            "end": {
              "pos": 29,
              "row": 2,
              "col": 6
            }
          },
          "comment": "// Numbers test\n",
          "start": {
            "pos": 16,
            "row": 1,
            "col": 0
          },
          "end": {
            "pos": 29,
            "row": 2,
            "col": 6
          }
        },
        "rvalue": {
          "kind": "NumberLiteral",
          "literal": "123e12",
          "value": "123000000000000",
          "start": {
            "pos": 31,
            "row": 3,
            "col": 0
          },
          "end": {
            "pos": 37,
            "row": 3,
            "col": 7
          }
        },
        "comment": null,
        "start": {
          "pos": 16,
          "row": 1,
          "col": 0
        },
        "end": {
          "pos": 37,
          "row": 3,
          "col": 7
        }
      },
      "rvalue": {
        "kind": "NumberLiteral",
        "literal": "123e+12",
        "value": "123000000000000",
        "start": {
          "pos": 39,
          "row": 4,
          "col": 0
        },
        "end": {
          "pos": 46,
          "row": 4,
          "col": 7
        }
      },
      "comment": null,
      "start": {
        "pos": 16,
        "row": 1,
        "col": 0
      },
      "end": {
        "pos": 46,
        "row": 4,
        "col": 7
      }
    },
    "rvalue": {
      "kind": "NumberLiteral",
      "literal": "123e-12",
      "value": 1.23e-10,
      "start": {
        "pos": 48,
        "row": 5,
        "col": 0
      },
      "end": {
        "pos": 55,
        "row": 5,
        "col": 7
      }
    },
    "comment": null,
    "start": {
      "pos": 16,
      "row": 1,
      "col": 0
    },
    "end": {
      "pos": 55,
      "row": 5,
      "col": 7
    }
  }
}